scripts/o3de/o3de/compatibility.py

```diff
@@ -0,0 +1,295 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+"""
+This file contains version utilities
+"""
+from packaging.version import Version, InvalidVersion
+from packaging.specifiers import SpecifierSet
+import pathlib
+import logging
+from o3de import manifest, utils, cmake
+
+logger = logging.getLogger('o3de.compatibility')
+logging.basicConfig(format=utils.LOG_FORMAT)
+
+def project_engine_compatible(project_path:pathlib.Path, project_json_data:dict, engine_json_data:dict) -> bool:
+    """
+    Returns True if the project is compatible with the provided engine data.
+    :param project_path: path to the project
+    :param project_json_data: project json data dictionary
+    :param engine_json_data: engine json data dictionary
+    """
+    if not o3de_object_is_compatible_with_engine(project_json_data, engine_json_data):
+        logger.error(f'{project_json_data["project_name"]} is not known to be compatible with the '
+                        f'engine {engine_json_data["engine_name"]}{engine_json_data.get("version","")} and requires the --force parameter '
+                        'to register.')
```

Comment:
What does the "registration" have to do with checking that the engine is compatible?

This part of the error message seems that it should in register.py

Reply:
Ah yes, I moved this logic from registery.py to here and didn't update the error logging correctly.  Will address.

Reply:
I think I'm going to refactor this code to return incompatible objects instead of True/False so I can provide better logging higher up